bam/bam_test.go

```diff
@@ -193,6 +194,34 @@ func BenchmarkRead(b *testing.B) {
 	f.Close()
 }
 
+func BenchmarkReadIndex(b *testing.B) {
+	if *findex == "" {
+		b.Skip("no index file specified")
+	}
+	b.StopTimer()
+	f, err := os.Open(*findex)
+	if err != nil {
+		b.Fatalf("Open failed: %v", err)
+	}
+
+	buf, err := ioutil.ReadAll(f)
+	if err != nil {
+		b.Fatalf("Index read failed: %v", err)
+	}
+	f.Close()
+
+	b.StartTimer()
+	for i := 0; i < b.N; i++ {
+
+		rdr := bytes.NewBuffer(buf)
```

Comment:
Instead of a bytes.Buffer in the loop, have a bytes.Reader that is seek(0, io.SeekStart)ed on each loop.

Please remove the blanks around the seek line.